interp/iostream_test.go

```diff
@@ -0,0 +1,73 @@
+package interp
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+)
+
+const (
+	outputStreamBufferSize = 1024
+)
+
+func TestStreamDoubleClose(t *testing.T) {
+	dir := t.TempDir()
+	t.Run("InFile", func(t *testing.T) {
+		f, err := os.Create(filepath.Join(dir, "infile"))
+		if err != nil {
+			t.Fatal(err)
+		}
+		in := newInFileStream(f)
+		checkDoubleClose(t, in)
+	})
+	t.Run("OutFile", func(t *testing.T) {
+		f, err := os.Create(filepath.Join(dir, "outfile"))
+		if err != nil {
+			t.Fatal(err)
+		}
+		out := newOutFileStream(f, outputStreamBufferSize)
+		checkDoubleClose(t, out)
+	})
+	t.Run("InCmd", func(t *testing.T) {
+		cmd := execDefShell("echo close me")
+		in, err := newInCmdStream(cmd)
+		if err != nil {
+			t.Fatal(err)
+		}
+		checkDoubleClose(t, in)
+	})
+	t.Run("OutCmd", func(t *testing.T) {
+		cmd := execDefShell("echo close me")
+		out, err := newOutCmdStream(cmd)
+		if err != nil {
+			t.Fatal(err)
+		}
+		checkDoubleClose(t, out)
+	})
+}
+
+func execDefShell(scriptlet string) *exec.Cmd {
```

Comment:
"def" sounds more like "define" to me in a programming context. Let's just spell this out:

```suggestion
func execDefaultShell(scriptlet string) *exec.Cmd {
```

Reply:
The verb already exists so it can't be "define". That would be "execute define shell" which is meaningless.

As you wish.